url026791d51b_dhleong_SeriesTracker_v6_tgz-pJ8-stv6_SeriesTrackerJ8/src/stv6/Profile.java

```diff
@@ -0,0 +1,801 @@
+		
+		if (p.dbtype.value(o).equals("mysql")) {
+			// try to build a mysql db
+			String dbserver = p.mysqlserver.value(o);
+			String dbname = p.mysqldb.value(o);
+			String dbuser = p.mysqluser.value(o);
+			String dbpass = p.mysqlpass.value(o);
+			db = new MysqlDatabase(dbserver, dbname, dbuser, dbpass);
+		} else if (p.dbtype.value(o).equals("flat")) {
+			//String cfgFile = p.flatDbArg.value(o);
+			//db = new FlatFileDatabase(cfgFile);
+			System.err.println("Flat-file database support is not yet available.");
+			System.exit(1);
+		} else {
+			// default to sqlite...?
+			db = new SqliteDatabase( p.dbFileArg.value(o) );
+		} 
+		
+		if (p.emtype.value(o).equals("mediatomb")) {
+			int mtport = p.mtportArg.value(o);
+			boolean checkTr = o.has("mediatomb-tr-check");
+			String trProf = p.mtTrProfile.value(o);
+			
+			epmgr = new MediaTombManager(STServer.getBroadcastingIp(), 
+					mtport, trProf, checkTr,
+					p.mtTrForce.values(o)
+			);
+		} else if (p.emtype.value(o).equals("local")) {
+			if (!o.has(p.localArg)) {
+				System.err.println("Error: Local episodes requested, but local-folders " +
+						"not provided");
+				System.exit(1);
+			}
+			epmgr = new FileSystemManager(p.localArg.value(o));
+		} else if (p.emtype.value(o).equals("pms")) {
+		    // ps3 media server
+		    
+		    epmgr = new UpnpManager(STServer.getBroadcastingIp(), true);
+		    System.out.println("Note: PMS must be running on the same computer as STv6");
+		} else if (p.emtype.value(o).equals("upnp")) {
+		    int upnpPort = p.upnpPortArg.value(o);
+		    
+		    if (o.has(p.localArg)) {
+	            epmgr = new UpnpManager(STServer.getBroadcastingIp(), upnpPort,
+	                    p.localArg.value(o));
+		    } else {
+		        epmgr = new UpnpManager(STServer.getBroadcastingIp(), upnpPort);
+		    }
+		    System.out.println("Warning: UPNP support is ALPHA, and NOT automatic!");
+		    System.out.println("The server must be local, and you must specify the port...");
+		} else {
+			// default to tversity
+			
+			int tvport = p.tvportArg.value(o);		
+			epmgr = new TversityManager(STServer.getBroadcastingIp(), tvport);
+			System.out.println("Warning: TVersity support is UNTESTED!");
+			System.out.println("It *might* work, though :)");
+		}
+	}
+	
+	@Override
+    public void reload() {
+		
+		// make sure we only try to reload once
+		synchronized(reloading) {
+			if (!reloading) {
+				reloading = true;
+				new Thread(this, "Profile-Reload").start();
+			}
+		}
+	}
+
+	/**
+	 * Threaded reloading fun
+	 */
+	@Override
+	public void run() {
+		STHandlerManager.getInstance().setStaticMessage(true);
+		StaticMessageHandler.getInstance().setBody();
+		System.out.print("Reloading... ");
+		if (!profileFromCommandLine && !isSelected() && profileFile == null) {
+			System.out.println("Failed (No profile chosen; will prompt)");
+			System.out.println("Server listening on port: " + getPort());
+			synchronized(reloading) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Object reloading synchronized on line 509 should not be re-assigned in the same synchronized block on line 510. After this re-assignment, you lose the reference to the synchronized object.